test/expect-tests/dune_file_watcher/dune_file_watcher_tests_linux.ml

```diff
@@ -69,4 +65,13 @@ let%expect_test _ =
     {|
   { path = In_source_tree "d/w/y"; kind = "Created" }
   { path = In_source_tree "d/w/y"; kind = "File_changed" }
-|}]
+  |}];
+  Dune_file_watcher.add_watch watcher (Path.of_string "e/1/2");
+  let (_ : _) = Fpath.mkdir_p "e/1" in
+  Stdio.Out_channel.write_all "e/1/2" ~data:"z";
+  print_events 3;
+  [%expect
+    {|
+    { path = In_source_tree "e"; kind = "Created" }
+    { path = In_source_tree "e/1"; kind = "Created" }
+    { path = In_source_tree "e/1/2"; kind = "Created" } |}]
```

Comment:
Could you also add a test where we move a whole tree (say, with 2 levels and 2 subdirectories in each node) into a watched place? I'd like to make sure we can correctly deal with moves and trees (in addition to creating sequences of nested paths as tested above).